2-collectors/scala-stream-collector/examples/config.hocon.sample

```diff
@@ -55,10 +55,21 @@ collector {
     name = {{collectorCookieName}}
     name = ${?COLLECTOR_COOKIE_NAME}
     # The domain is optional and will make the cookie accessible to other
-    # applications on the domain. Comment out this line to tie cookies to
-    # the collector's full domain
-    domain = "{{cookieDomain}}"
-    domain = ${?COLLECTOR_COOKIE_DOMAIN}
+    # applications on the domain. Comment out these lines to tie cookies to
+    # the collector's full domain.
+    # The domain is determined by matching the origin domains (from the Origin header of the request)
+    # to the list below. The first match is used. If no matches are found, the value will not be set
+    # and the collector's domain will be used.
+    # If the Origin header contains any subdomains, only the root domain names will be matched against
+    # the list below. That way, the cookie will be readable from all subdomains on the root domain.
+    domains = [
+        "{{cookieDomain1}}"
+        "{{cookieDomain2}}"
+        # ... more domains
+    ]
```

Comment:
I think the original issue is around a mapping not just a list.

e.g. you get an incoming request with a `snplow.com` origin header, and you want to send back `snowplowanalytics.com` in your cookie.

Also, having given it a bit more thought, I think we need a `fallbackDomain` which is the mirroring of the current mechanism so that it is easier to transition.

Reply:
After chatting with @yalisassoon , he confirmed that a list of domains is enough. The only reason we would need a mapping would be to tie subdomains to the main domain. However, I have opted to instead allow a `*` wildcard to be used:

```
domains = [
  domain.com
  *.domain.com
]
```

The first one will match the main domain and the second one will match any subdomain on it.

@BenFradet could you expand on your other thought about the fallback? Do you mean that we need a default domain that will always be used, even if there are no matches from the Origin hosts?

Reply:
>After chatting with @yalisassoon ...

then I think we need to modify the ticket name

>@BenFradet could you expand on your other thought about the fallback? Do you mean that we need a default domain that will always be used, even if there are no matches from the Origin hosts?

yes

Reply:
Done and done.